diff -Nuar pisi_/pxml/autoxml.py pisi/pxml/autoxml.py
--- pisi_/pxml/autoxml.py	2011-08-21 15:30:29.716000065 +0300
+++ pisi/pxml/autoxml.py	2011-08-21 15:31:02.372000065 +0300
@@ -338,7 +338,7 @@
         cls.__init__ = initialize
 
         cls.decoders = decoders
-        def decode(self, node, errs, where = unicode(cls.tag), use_ondemand = False):
+        def decode(self, node, errs, where = unicode(cls.tag), use_ondemand = True):
             self.use_ondemand = use_ondemand
             for base in cls.autoxml_bases:
                 base.decode(self, node, errs, where, use_ondemand)
@@ -421,7 +421,7 @@
             cls.__ne__ = notequal
 
         if xmlfile_support:
-            def parse(self, xml, keepDoc = False, use_ondemand = False):
+            def parse(self, xml, keepDoc = False, use_ondemand = True):
                 "parse XML string and decode it into a python object"
                 self.use_ondemand = use_ondemand
                 self.parsexml(xml)
@@ -442,7 +442,7 @@
 
             def read(self, uri, keepDoc = False, tmpDir = '/tmp',
                      sha1sum = False, compress = None, sign = None, copylocal = False, nodecode = False,
-                     use_ondemand = False):
+                     use_ondemand = True):
                 "read XML file and decode it into a python object"
                 self.use_ondemand = use_ondemand
 
