Index: Bugzilla/User.pm
===================================================================
--- Bugzilla/User.pm	(revision 27740)
+++ Bugzilla/User.pm	(revision 27741)
@@ -667,7 +667,7 @@
             $query .= " AND group_control_map.membercontrol = " . CONTROLMAPMANDATORY;
         }
         $query .= "     AND group_id NOT IN(" . $self->groups_as_string . ") " .
-                  "   WHERE group_id IS NULL " .
+                  "   WHERE group_id IS NULL  and products.disallownew = 0 " .
                   "ORDER BY name";
 
         my $prod_ids = Bugzilla->dbh->selectcol_arrayref($query);
