diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/custom/bug/create/comment.txt.tmpl en/custom/bug/create/comment.txt.tmpl
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/custom/bug/create/comment.txt.tmpl	1970-01-01 00:00:00.000000000 +0000
+++ en/custom/bug/create/comment.txt.tmpl	2010-08-10 09:16:13.539326369 +0000
@@ -0,0 +1,52 @@
+[%# The contents of this file are subject to the Mozilla Public
+  # License Version 1.1 (the "License"); you may not use this file
+  # except in compliance with the License. You may obtain a copy of
+  # the License at http://www.mozilla.org/MPL/
+  #
+  # Software distributed under the License is distributed on an "AS
+  # IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
+  # implied. See the License for the specific language governing
+  # rights and limitations under the License.
+  #
+  # The Original Code is the Bugzilla Bug Tracking System.
+  #
+  # The Initial Developer of the Original Code is Netscape Communications
+  # Corporation. Portions created by Netscape are
+  # Copyright (C) 1998 Netscape Communications Corporation. All
+  # Rights Reserved.
+  #
+  # Contributor(s): Gervase Markham <gerv@gerv.net>
+  #%]
+[%# INTERFACE:
+  # This template has no interface.
+  #
+  # Form variables from a bug submission (i.e. the fields on a template from
+  # enter_bug.cgi) can be access via Bugzilla.cgi.param. It can be used to
+  # pull out various custom fields and format an initial Description entry
+  # from them.
+  #%]  
+[% USE Bugzilla %]
+[% cgi = Bugzilla.cgi %]
+
+[%+ cgi.param("comment") IF cgi.param("comment") %]
+
+[%+ IF cgi.param("reproducible") != "--" -%]
+Reproducible: [%+ cgi.param("reproducible") %]
+[% END %]
+
+[% IF !(cgi.param("reproduce_steps").match('^1\.\s*2\.\s*3\.\s*$') || cgi.param("reproduce_steps").match('^\s*$')) %]
+Steps to Reproduce:
+[%+ cgi.param("reproduce_steps") %]
+[% END %]
+
+[% IF cgi.param("actual_results") -%]
+Actual Results:  
+[%+ cgi.param("actual_results") %]
+[% END %]
+
+[% IF cgi.param("expected_results") %]
+Expected Results:  
+[%+ cgi.param("expected_results") %]
+[% END %]
+
+[%+ cgi.param("additional_info") %]
diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/custom/bug/create/create.html.tmpl en/custom/bug/create/create.html.tmpl
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/custom/bug/create/create.html.tmpl	1970-01-01 00:00:00.000000000 +0000
+++ en/custom/bug/create/create.html.tmpl	2010-08-10 09:16:13.539326369 +0000
@@ -0,0 +1,824 @@
+[%# The contents of this file are subject to the Mozilla Public
+  # License Version 1.1 (the "License"); you may not use this file
+  # except in compliance with the License. You may obtain a copy of
+  # the License at http://www.mozilla.org/MPL/
+  #
+  # Software distributed under the License is distributed on an "AS
+  # IS" basis, WITHOUT WARRANTY OF ANY KIND, either express or
+  # implied. See the License for the specific language governing
+  # rights and limitations under the License.
+  #
+  # The Original Code is the Bugzilla Bug Tracking System.
+  #
+  # The Initial Developer of the Original Code is Netscape Communications
+  # Corporation. Portions created by Netscape are
+  # Copyright (C) 1998 Netscape Communications Corporation. All
+  # Rights Reserved.
+  #
+  # Contributor(s): Gervase Markham <gerv@gerv.net>
+  #                 Ville Skyttä <ville.skytta@iki.fi>
+  #                 Shane H. W. Travis <travis@sedsystems.ca>
+  #                 Marc Schumann <wurblzap@gmail.com>
+  #                 Akamai Technologies <bugzilla-dev@akamai.com>
+  #                 Max Kanat-Alexander <mkanat@bugzilla.org>
+  #                 Frédéric Buclin <LpSolit@gmail.com>
+  #%]
+
+[% PROCESS "global/field-descs.none.tmpl" %]
+
+[% title = BLOCK %] Enter [% terms.Bug %] : [% product.name FILTER html %][% END %]
+
+[% PROCESS global/header.html.tmpl
+  title = title
+  style_urls = [ 'skins/standard/create_attachment.css',
+                 'skins/standard/yui/calendar.css' ]
+  javascript_urls = [ "js/attachment.js", "js/util.js", "js/yui/calendar.js",
+                      "js/field.js", "js/yui/cookie.js", "js/TUI.js" ]
+%]
+
+[% tablecolour = "#FFFFCC" %]
+<script type="text/javascript">
+<!--
+
+var initialowners = new Array([% product.components.size %]);
+var last_initialowner;
+var initialccs = new Array([% product.components.size %]);
+var components = new Array([% product.components.size %]);
+var comp_desc = new Array([% product.components.size %]);
+var flags = new Array([% product.components.size %]);
+[% IF Param("useqacontact") %]
+    var initialqacontacts = new Array([% product.components.size %]);
+    var last_initialqacontact;
+[% END %]
+[% count = 0 %]
+[%- FOREACH c = product.components %]
+    components[[% count %]] = "[% c.name FILTER js %]";
+    comp_desc[[% count %]] = "[% c.description FILTER html_light FILTER js %]";
+    initialowners[[% count %]] = "[% c.default_assignee.login FILTER js %]";
+    [% flag_list = [] %]
+    [% FOREACH f = c.flag_types.bug %]
+      [% NEXT UNLESS f.is_active %]
+      [% flag_list.push(f.id) %]
+    [% END %]
+    [% FOREACH f = c.flag_types.attachment %]
+      [% NEXT UNLESS f.is_active %]
+      [% flag_list.push(f.id) %]
+    [% END %]
+    flags[[% count %]] = [[% flag_list.join(",") FILTER js %]];
+    [% IF Param("useqacontact") %]
+        initialqacontacts[[% count %]] = "[% c.default_qa_contact.login FILTER js %]";
+    [% END %]
+
+    [% SET initial_cc_list = [] %]
+    [% FOREACH cc_user = c.initial_cc %]
+      [% initial_cc_list.push(cc_user.login) %]
+    [% END %]
+    initialccs[[% count %]] = "[% initial_cc_list.join(', ') FILTER js %]";
+
+    [% count = count + 1 %]
+[%- END %]
+
+function set_assign_to() {
+    // Based on the selected component, fill the "Assign To:" field
+    // with the default component owner, and the "QA Contact:" field
+    // with the default QA Contact. It also selectively enables flags.
+    var form = document.Create;
+    var assigned_to = form.assigned_to.value;
+
+[% IF Param("useqacontact") %]
+    var qa_contact = form.qa_contact.value;
+[% END %]
+
+    var index = -1;
+    if (form.component.type == 'select-one') {
+        index = form.component.selectedIndex;
+    } else if (form.component.type == 'hidden') {
+        // Assume there is only one component in the list
+        index = 0;
+    }
+    if (index != -1) {
+        var owner = initialowners[index];
+        var component = components[index];
+        if (assigned_to == last_initialowner
+            || assigned_to == owner
+            || assigned_to == '') {
+            form.assigned_to.value = owner;
+            last_initialowner = owner;
+        }
+
+        document.getElementById('initial_cc').innerHTML = initialccs[index];
+        document.getElementById('comp_desc').innerHTML = comp_desc[index];
+
+        [% IF Param("useqacontact") %]
+            var contact = initialqacontacts[index];
+            if (qa_contact == last_initialqacontact
+                || qa_contact == contact
+                || qa_contact == '') {
+                  form.qa_contact.value = contact;
+                  last_initialqacontact = contact;
+            }
+        [% END %]
+
+        // First, we disable all flags. Then we re-enable those
+        // which are available for the selected component.
+        var inputElements = document.getElementsByTagName("select");
+        var inputElement, flagField;
+        for ( var i=0 ; i<inputElements.length ; i++ ) {
+            inputElement = inputElements.item(i);
+            if (inputElement.name.search(/^flag_type-(\d+)$/) != -1) {
+                var id = inputElement.name.replace(/^flag_type-(\d+)$/, "$1");
+                inputElement.disabled = true;
+                // Also disable the requestee field, if it exists.
+                inputElement = document.getElementById("requestee_type-" + id);
+                if (inputElement) inputElement.disabled = true;
+            }
+        }
+        // Now enable flags available for the selected component.
+        for (var i = 0; i < flags[index].length; i++) {
+            flagField = document.getElementById("flag_type-" + flags[index][i]);
+            // Do not enable flags the user cannot set nor request.
+            if (flagField && flagField.options.length > 1) {
+                flagField.disabled = false;
+                // Re-enabling the requestee field depends on the status
+                // of the flag.
+                toggleRequesteeField(flagField, 1);
+            }
+        }
+    }
+}
+
+function handleWantsAttachment(wants_attachment) {
+    if (wants_attachment) {
+        document.getElementById('attachment_false').style.display = 'none';
+        document.getElementById('attachment_true').style.display = 'block';
+    }
+    else {
+        document.getElementById('attachment_false').style.display = 'block';
+        document.getElementById('attachment_true').style.display = 'none';
+        clearAttachmentFields();
+    }
+}
+
+
+TUI_alternates['expert_fields'] = 'Show Advanced Fields';
+// Hide the Advanced Fields by default, unless the user has a cookie
+// that specifies otherwise.
+TUI_hide_default('expert_fields');
+
+-->
+</script>
+
+[% USE Bugzilla %]
+[% SET select_fields = {} %]
+[% FOREACH field = Bugzilla.get_fields(
+  { type => constants.FIELD_TYPE_SINGLE_SELECT, custom => 0 })
+%]
+  [% select_fields.${field.name} = field %]
+[% END %]
+
+<form name="Create" id="Create" method="post" action="post_bug.cgi"
+      enctype="multipart/form-data">
+<input type="hidden" name="product" value="[% product.name FILTER html %]">
+<input type="hidden" name="token" value="[% token FILTER html %]">
+
+<table valign="top" cellpadding="5" cellspacing="5" border="0">
+<tbody>
+  <tr>
+    <td colspan="4">
+    [%# Migration note: The following file corresponds to the old Param
+      # 'entryheaderhtml'
+      #%]
+    [% INCLUDE 'bug/create/user-message.html.tmpl' %]
+    </td>
+  </tr>
+
+  <tr>
+    <td colspan="4">
+        <a id="expert_fields_controller" class="controller bz_default_hidden"
+            href="javascript:TUI_toggle_class('expert_fields')">Hide Advanced Fields</a>
+      [%# Show the link if the browser supports JS %]
+      <script type="text/javascript">
+          YAHOO.util.Dom.removeClass('expert_fields_controller', 
+                                     'bz_default_hidden');
+      </script>
+    </td>
+  </tr>
+
+  <tr bgcolor="[% tablecolour %]">
+    <td align="right" valign="top">
+    <b>Product</b>
+    </td>
+    <td valign="top">[% product.name FILTER html %]</td>
+  </tr>
+
+  [%# We can't use the select block in these two cases for various reasons. %]
+  <tr>
+    <td align="right" valign="top">
+      <a href="describecomponents.cgi?product=[% product.name FILTER url_quote %]">
+        <b>Component</b>
+        </a>
+    </td>
+    <td valign="top">
+    <table border="0" cellpadding="0" cellspacing="0">
+    <tr>
+    <td valign="top">
+      <select name="component" onchange="set_assign_to();" size="7">
+        [%# Build the lists of assignees and QA contacts if "usemenuforusers" is enabled. %]
+        [% IF Param("usemenuforusers") %]
+          [% assignees_list = user.get_userlist.clone %]
+          [% qa_contacts_list = user.get_userlist.clone %]
+        [% END %]
+
+        [%- FOREACH c = product.components %]
+          <option value="[% c.name FILTER html %]"
+            [% " selected=\"selected\"" IF c.name == default.component_ %]>
+            [% c.name FILTER html -%]
+          </option>
+          [% IF Param("usemenuforusers") %]
+            [% INCLUDE build_userlist default_user = c.default_assignee,
+                                      userlist = assignees_list %]
+            [% INCLUDE build_userlist default_user = c.default_qa_contact,
+                                      userlist = qa_contacts_list %]
+          [% END %]
+        [%- END %]
+      </select>
+    </td>
+
+    <td valign="top" width="100%">
+      [%# Enclose the fieldset in a nested table so that its width changes based
+        # on the length on the component description. %]
+      <table>
+        <tr>
+          <td>
+            <fieldset>
+              <legend>Component Description</legend>
+              <div id="comp_desc" class="comment">Select a component to read its description.</div>
+            </fieldset>
+          </td>
+        </tr>
+      </table>
+    </td>
+    </tr>
+    </table>
+    <p>
+    The area which the problem occured.
+    You can follow, <a target="_blank" href="describecomponents.cgi?product=[% product.name FILTER url_quote %]">component description</a> link for help.
+    </p>
+    </td>
+  </tr>
+
+  <tr>
+    [% INCLUDE bug/field.html.tmpl
+      bug = default, field = select_fields.rep_platform, editable = 1,
+      value = default.rep_platform %]
+  </tr>
+
+ <tr>
+    <td align="right" valign="top">
+    <b>Version</b>
+    </td>
+    <td>
+      <select name="version" size="5">
+        [%- FOREACH v = version %]
+          <option value="[% v FILTER html %]"
+            [% ' selected="selected"' IF v == default.version %]>[% v FILTER html -%]
+          </option>
+        [%- END %]
+      </select>
+    </td>
+  </tr>
+</tbody>
+
+<tbody class="expert_fields">
+  <tr>
+    [% IF Param('usetargetmilestone') && Param('letsubmitterchoosemilestone') %]
+      [% sel = { description => 'Target Milestone', name => 'target_milestone' } %]
+      [% INCLUDE select %]
+    [% ELSE %]
+      <!--td colspan="2">&nbsp;</td-->
+    [% END %]
+
+    [% IF Param('letsubmitterchoosepriority') %]
+      [% INCLUDE bug/field.html.tmpl
+        bug = default, field = select_fields.priority, editable = 1, 
+        value = default.priority %]
+    [% ELSE %]
+      <!--td colspan="2">&nbsp;</td-->
+    [% END %]
+  </tr>
+</tbody>
+
+<tbody class="expert_fields">
+  <tr>
+    <td colspan="4">&nbsp;</td>
+  </tr>
+
+  <tr>
+[% IF bug_status.size <= 1 %]
+  <input type="hidden" name="bug_status" 
+         value="[% default.bug_status FILTER html %]">
+     <td align="right" valign="top">
+         <b>State</b>
+    </td>
+    <td>[% get_status(default.bug_status) FILTER html %]</td>
+[% ELSE %]
+    [% sel = { description => 'State', name => 'bug_status' } %]
+    [% INCLUDE select %]
+[% END %]
+
+    <td>&nbsp;</td>
+    [%# Calculate the number of rows we can use for flags %]
+    [% num_rows = 6 + (Param("useqacontact") ? 1 : 0) +
+                      (user.in_group(Param('timetrackinggroup')) ? 3 : 0) +
+                      (Param("usebugaliases") ? 1 : 0)
+    %]
+
+    <td rowspan="[% num_rows FILTER html %]">
+      [% IF product.flag_types.bug.size > 0 %]
+        [% display_flag_headers = 0 %]
+        [% any_flags_requesteeble = 0 %]
+
+        [% FOREACH flag_type = product.flag_types.bug %]
+          [% NEXT UNLESS flag_type.is_active %]
+          [% display_flag_headers = 1 %]
+          [% SET any_flags_requesteeble = 1 IF flag_type.is_requestable && flag_type.is_requesteeble %]
+        [% END %]
+
+        [% IF display_flag_headers %]
+          [% PROCESS "flag/list.html.tmpl" flag_types = product.flag_types.bug
+                                           any_flags_requesteeble = any_flags_requesteeble
+                                           flag_table_id = "bug_flags"
+          %]
+        [% END %]
+      [% END %]
+    </td>
+  </tr>
+
+  <tr>
+  <td align="right" valign="top">
+      <b><a href="page.cgi?id=fields.html#assigned_to">Assigned to</a></b>
+  </td>
+    <td colspan="2">
+    [% IF product.name == "Review" %]
+     [% INCLUDE global/userselect.html.tmpl
+        name => "assigned_to"
+        value => user.login
+        disabled => assigned_to_disabled
+        size => 30
+        emptyok => 1
+        custom_userlist => assignees_list
+      %]
+     [% ELSE %]
+     [% INCLUDE global/userselect.html.tmpl
+        name => "assigned_to"
+        value => assigned_to
+        disabled => assigned_to_disabled
+        size => 30
+        emptyok => 1
+        custom_userlist => assignees_list
+      %]
+      [% END %]
+      <noscript>(Leave blank to assign to component's default assignee)</noscript>
+    </td>
+  </tr>
+
+[% IF Param("useqacontact") %]
+    <tr>
+        <td align="right" valign="top">
+            <b>QA Contact</b>
+        </td>
+      <td colspan="2">
+      [% INCLUDE global/userselect.html.tmpl
+         name => "qa_contact"
+         value => qa_contact
+         disabled => qa_contact_disabled
+         size => 30
+         emptyok => 1
+         custom_userlist => qa_contacts_list
+       %]
+        <noscript>(Leave blank to assign to default qa contact)</noscript>
+      </td>
+    </tr>
+[% END %]
+
+  <tr>
+    <td align="right" valign="top">
+        <b>CC</b>
+    </td>
+
+    <td colspan="2">
+      [% INCLUDE global/userselect.html.tmpl
+         name => "cc"
+         value => cc
+         disabled => cc_disabled
+         size => 30
+         multiple => 5
+       %]
+    </td>
+  </tr>
+
+  <tr>
+    <td align="right" valign="top">
+    <b>Default CC</b>
+    </td>
+    <td colspan="2">
+      <div id="initial_cc">
+          <!-- This has to happen after everything above renders,
+               and onload doesn't work. So this is as good a place
+               as any to put it. -->
+          <script type="text/javascript">set_assign_to();</script>
+      </div>
+   </td>
+  </tr>
+
+  <tr>
+    <td colspan="3">&nbsp;</td>
+  </tr>
+
+[% IF Param("usebugaliases") %]
+  <tr>
+    <th>Alias:</th>
+    <td colspan="2">
+      <input name="alias" size="20">
+    </td>
+  </tr>
+[% END %]
+
+  <tr>
+    <th>URL</th>
+    <td colspan="2">
+      <input name="bug_file_loc" size="40"
+             value="[% bug_file_loc FILTER html %]">
+    </td>
+  </tr>
+</tbody>
+
+<tbody class="expert_fields">
+  [% USE Bugzilla %]
+
+  [% FOREACH field = Bugzilla.active_custom_fields %]
+    [% NEXT UNLESS field.enter_bug %]
+    [% SET value = ${field.name}.defined ? ${field.name} : "" %]
+    <tr>
+      [% INCLUDE bug/field.html.tmpl 
+        bug = default, field = field, value = value, editable = 1, 
+        value_span = 3 %]
+    </tr>
+  [% END %]
+</tbody>
+
+<tbody>
+
+  <tr bgcolor="[% tablecolour %]">
+      <td align="right" valign="top">
+        <b>Summary</b>
+    </td>
+    <td valign="top">
+      <input type="text" size="70" name="short_desc" id="short_desc"·value="[% short_desc FILTER html %]"  maxlength="255" spellcheck="true">
+      <p>
+        <i> A sentence which summarises the problem.
+            Please be descriptive and use lots of keywords.</i>
+      </p>
+      <p>
+        <tt>
+          <font color="#990000">Bad Example</font>: KDE crashed.
+        </tt>
+        <br>
+        <tt>
+          <font color="#009900">Good Example</font>:
+          KDE crashed when a directory tried to open with Dolphin.
+        </tt>
+      </p>
+    </td>
+  </tr>
+
+  <tr>
+    <td align="right" valign="top">
+      <b>Details</b>
+    </td>
+    <td valign="top">
+      [% INCLUDE global/textarea.html.tmpl
+        name    = 'comment'
+        minrows = 6
+        cols    = constants.COMMENT_COLS
+      %]
+      <p>
+        <i>Expand on the Summary. Please be as specific as possible about what is wrong.</i>
+      </p>
+      <p>
+        <tt>
+          <font color="#990000">Bad Example</font>: KDE crashed. You suck!
+        </tt>
+        <br>
+        <tt>
+          <font color="#009900">Bad Example</font>: When a directory which contains a lot of files is opened with Dolphin, Dolphin hangs and after a while KDE crashes.
+        </tt>
+      </p>
+    </td>
+  </tr>
+
+ <tr bgcolor="[% tablecolour %]">
+    <td align="right" valign="top">
+      <b>Reproducibility</b>
+    </td>
+    <td valign="top">
+      <select name="reproducible">
+        <option name="null" value="--">
+        --
+        <option name="AlwaysReproducible" value="Always">
+          Happens every time.
+        </option>
+        <option name="Sometimes" value="Sometimes">
+          Happens sometimes, but not always.
+        </option>
+        <option name="DidntTry" value="Didn't try">
+           Haven't tried to reproduce it.
+        </option>
+        <option name="NotReproducible" value="Couldn't Reproduce">
+          Tried, but couldn't reproduce it.
+        </option>
+      </select>
+    </td>
+  </tr>
+  <tr>
+    <td align="right" valign="top">
+      <b>Steps to Reproduce</b>
+    </td>
+    <td valign="top">
+      [% INCLUDE global/textarea.html.tmpl
+        name           = 'reproduce_steps'
+        minrows        = 4
+        cols           = constants.COMMENT_COLS
+        defaultcontent = "1.\n2.\n3."
+      %]
+      <p>
+        <i>Describe how to reproduce the problem, step by
+        step. Include any special setup steps.</i>
+      </p>
+    </td>
+  </tr>
+
+ <tr bgcolor="[% tablecolour %]">
+    <td valign="top" align="right">
+      <b>Actual Results</b>
+    </td>
+    <td valign="top">
+      [% INCLUDE global/textarea.html.tmpl
+        name    = 'actual_results'
+        minrows = 4
+        cols    = constants.COMMENT_COLS
+      %]
+      <p>
+       <i>What happened after you performed the steps above?</i>
+      </p>
+    </td>
+  </tr>
+
+
+ <tr>
+    <td valign="top" align="right">
+      <b>Expected Results</b>
+    </td>
+    <td valign="top">
+      [% INCLUDE global/textarea.html.tmpl
+        name    = 'expected_results'
+        minrows = 4
+        cols    = constants.COMMENT_COLS
+      %]
+      <p>
+        <i>What should the software have done instead?</i>
+     </p>
+    </td>
+  </tr>
+ <tr bgcolor="[% tablecolour %]">
+    <td valign="top" align="right">
+      <b>Additional Information</b>
+    </td>
+    <td valign="top">
+      [% INCLUDE global/textarea.html.tmpl
+        name    = 'additional_info'
+        minrows = 8
+        cols    = constants.COMMENT_COLS
+      %]
+      <p>
+     <tt>
+    Add any additional information you feel may be relevant to this [% terms.Bug %], such as your sys     tem information, stack trace, or a testcase. Any information longer than a few lines, should be      added using the <b>Add an Attachment</b> link on the [% terms.Bug %], after
+    </tt>
+      </p>
+   </td>
+  </tr>
+
+ <tr>
+    <td valign="top" align="right">
+      <b>Severity</b>
+    </td>
+    <td valign="top">
+      <select name="bug_severity" size="5">
+        <option name="blocking" value="blocking">
+         Urgent: The bug makes whole system unusable.
+        </option>
+        <option name="high" value="high">
+          High: The bug makes the program unusable.
+        </option>
+        <option name="normal" value="normal" selected="selected">
+        Normal: The bug which makes a part of the program unusable.
+        </option>
+        <option name="low" value="low">
+        Low: A cosmetic problem, such as a misspelled word or misaligned text or an enhancement.
+        </option>
+        <option name="newfeature" value="newfeature">
+        New Feature: A new feature
+        </option>
+      </select>
+      <p>
+        <i>   Say how serious the problem is, or if your [% terms.Bug %] is a·
+        request for a new feature.</i>
+      </p>
+    </td>
+  </tr>
+
+  [% IF Param("insidergroup") && user.in_group(Param("insidergroup")) %]
+    <tr class="expert_fields">
+      <th>&nbsp;</th>
+      <td colspan="3">
+        &nbsp;&nbsp;
+        <input type="checkbox" id="commentprivacy" name="commentprivacy"
+          [% " checked=\"checked\"" IF commentprivacy %]>
+        <label for="commentprivacy">
+          Başlangıç Açıklaması Gizlidir
+        </label>
+      </td>
+    </tr>
+  [% END %]
+
+  <tr>
+    <th>Attachment</th>
+    <td colspan="3">
+      <script type="text/javascript">
+        <!--
+        document.write( '<div id="attachment_false">'
+                      +   '<input type="button" value="Add an attachment" '
+                      +          'onClick="handleWantsAttachment(true)"> '
+                      +   '<em style="display: none">This button has no '
+                      +   'functionality for you because your browser does '
+                      +   'not support CSS or does not use it.<\/em>'
+                      + '<\/div>'
+                      + '<div id="attachment_true" style="display: none">'
+                      +   '<input type="button" '
+                      +          'value="Don\'t add an attachment " '
+                      +          'onClick="handleWantsAttachment(false)">');
+        //-->
+      </script>
+        <fieldset>
+          <legend>Add an attachment</legend>
+          <table class="attachment_entry">
+            [% PROCESS attachment/createformcontents.html.tmpl
+                       flag_types = product.flag_types.attachment
+                       any_flags_requesteeble = 1
+                       flag_table_id ="attachment_flags" %]
+          </table>
+        </fieldset>
+      <script type="text/javascript">
+        <!--
+        document.write('<\/div>');
+        //-->
+      </script>
+    </td>
+  </tr>
+</tbody>
+
+<tbody class="expert_fields">
+  [% IF user.in_group('editbugs', product.id) %]
+    [% IF use_keywords %]
+      <tr>
+        <th><a href="describekeywords.cgi">Keywords</a></th>
+        <td colspan="3">
+          <input id="keywords" name="keywords" size="40"
+                 value="[% keywords FILTER html %]"> (isteğe bağlı)
+        </td>
+      </tr>
+    [% END %]
+
+    <tr>
+      <th>Depends on</th>
+      <td colspan="3">
+        <input name="dependson" accesskey="d" value="[% dependson FILTER html %]">
+      </td>
+    </tr>
+    <tr>
+      <th>Blocks</th>
+      <td colspan="3">
+        <input name="blocked" accesskey="b" value="[% blocked FILTER html %]">
+      </td>
+    </tr>
+  [% END %]
+</tbody>
+
+<tbody class="expert_fields">
+  [% IF group.size %]
+  <tr>
+    <th>&nbsp;</th>
+    <td colspan="3">
+      <br>
+        <strong>
+         Only users in all of the selected groups can view this [% terms.bug %]:
+        </strong>
+      <br>
+      <font size="-1">
+      (Leave all boxes unchecked to make this a public [% terms.bug %].)
+      </font>
+      <br>
+      <br>
+
+      <!-- Checkboxes -->
+      [% FOREACH g = group %]
+        &nbsp;&nbsp;&nbsp;&nbsp;
+        <input type="checkbox" id="bit-[% g.bit %]"
+          name="bit-[% g.bit %]" value="1"
+          [% " checked=\"checked\"" IF g.checked %]>
+          <label for="bit-[% g.bit %]">Güvenlik Hataları Üzerinde değişiklik yapabilir.</label><br>
+      [% END %]
+    </td>
+  </tr>
+  [% END %]
+</tbody>
+
+<tbody>
+  [%# Form controls for entering additional data about the bug being created. %]
+  [% Hook.process("form") %]
+
+  <tr>
+    <th>&nbsp;</th>
+    <td colspan="3">
+      <input type="submit" id="commit" value="Commit"
+             onclick="if (this.form.short_desc.value == '')
+             { alert('Bu hata için lütfen bir özet cümlesi giriniz!');
+               return false; } return true;">
+      &nbsp;&nbsp;&nbsp;&nbsp;
+      <input type="submit" name="maketemplate" id="maketemplate"
+             value="Değerleri yer imi olarak hatırlat"
+             class="expert_fields">
+    </td>
+  </tr>
+</tbody>
+  </table>
+  <input type="hidden" name="form_name" value="enter_bug">
+</form>
+
+[%# Links or content with more information about the bug being created. %]
+[% Hook.process("end") %]
+
+[% PROCESS global/footer.html.tmpl %]
+
+[%############################################################################%]
+[%# Block for SELECT fields                                                  #%]
+[%############################################################################%]
+
+[% BLOCK select %]
+  [% IF sel.description %]
+    <th>
+      <a href="page.cgi?id=fields.html#[% sel.name %]">[% sel.description %]</a>:
+    </th>
+  [% END %]
+
+  <td>
+    <select name="[% sel.name %]" id="[% sel.name %]">
+    [%- FOREACH x = ${sel.name} %]
+      <option value="[% x FILTER html %]"
+        [% " selected=\"selected\"" IF x == default.${sel.name} %]>
+        [% IF sel.name == "bug_status" %]
+          [% get_status(x) FILTER html %]
+        [% ELSE %]
+          [% x FILTER html %]
+        [% END %]</option>
+    [% END %]
+    </select>
+
+    [% IF sel.name == "bug_status" %]
+       <script type="text/javascript">
+        <!--
+          [%+ INCLUDE "bug/field-events.js.tmpl" 
+                     field = select_fields.bug_status %]
+        //-->
+        </script>
+    [% END %]
+  </td>
+[% END %]
+
+[% BLOCK build_userlist %]
+  [% user_found = 0 %]
+  [% default_login = default_user.login %]
+  [% RETURN UNLESS default_login %]
+
+  [% FOREACH user = userlist %]
+    [% IF user.login == default_login %]
+      [% user_found = 1 %]
+      [% LAST %]
+    [% END %]
+  [% END %]
+
+  [% userlist.push({login    => default_login,
+                    identity => default_user.identity,
+                    visible  => 1})
+     UNLESS user_found %]
+[% END %]
diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/.cvsignore en/.cvsignore
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/.cvsignore	2002-04-24 07:29:49.000000000 +0000
+++ en/.cvsignore	1970-01-01 00:00:00.000000000 +0000
@@ -1,2 +0,0 @@
-.htaccess
-custom
diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/bug/create/create.html.tmpl en/default/bug/create/create.html.tmpl
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/bug/create/create.html.tmpl	2009-02-12 00:46:54.000000000 +0000
+++ en/default/bug/create/create.html.tmpl	2010-08-10 09:16:13.272326684 +0000
@@ -36,6 +36,8 @@
                       "js/field.js", "js/yui/cookie.js", "js/TUI.js" ]
 %]
 
+[% tablecolour = "#FFFFCC" %]
+
 <script type="text/javascript">
 <!--
 
diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/bug/edit.html.tmpl en/default/bug/edit.html.tmpl
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/bug/edit.html.tmpl	2010-02-18 19:19:16.000000000 +0000
+++ en/default/bug/edit.html.tmpl	2010-08-10 09:16:13.518449624 +0000
@@ -417,10 +417,10 @@
             bug = bug, field = select_fields.rep_platform,
             no_tds = 1, value = bug.rep_platform
             editable = bug.check_can_change_field('rep_platform', 0, 1) %]
-       [%+ INCLUDE bug/field.html.tmpl 
+       <!--[%+ INCLUDE bug/field.html.tmpl 
             bug = bug, field = select_fields.op_sys, 
             no_tds = 1, value = bug.op_sys
-            editable = bug.check_can_change_field('op_sys', 0, 1) %]
+            editable = bug.check_can_change_field('op_sys', 0, 1) %]-->
        <script type="text/javascript">
          assignToDefaultOnChange(['product', 'component']);
        </script>
@@ -1034,7 +1034,7 @@
 [%############################################################################%]
 
 [% BLOCK section_timetracking %]
-  <table class="bz_time_tracking_table">
+  <!--table class="bz_time_tracking_table">
     <tr>
       <th>
         <label for="estimated_time">Orig. Est.</label>
@@ -1101,7 +1101,7 @@
         blocking this [% terms.bug %])</a>
       </td>
     </tr>
-  </table> 
+  </table--> 
 [% END %]
 
 [%############################################################################%]
diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/global/choose-product.html.tmpl en/default/global/choose-product.html.tmpl
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/global/choose-product.html.tmpl	2007-08-20 18:25:04.000000000 +0000
+++ en/default/global/choose-product.html.tmpl	2010-08-10 09:16:13.061449852 +0000
@@ -40,7 +40,7 @@
 [% DEFAULT title = "Choose a Product" %]
 [% PROCESS global/header.html.tmpl %]
 
-<table>
+<table align="center">
 
 [% FOREACH c = classifications %]
   [% IF c.object %]
@@ -52,14 +52,29 @@
 
   [% FOREACH p = c.products %]
     <tr>
-      <th align="right" valign="top">
+      <td valign="top">
+      <table>
+      <tr>
+      <td rowspan="3">
+        <a href="[% target %]?product=[% p.name FILTER url_quote -%]
+              [%- IF cloned_bug_id %]&amp;cloned_bug_id=[% cloned_bug_id FILTER url_quote %][% END -%] 
+              [%- IF format %]&amp;format=[% format FILTER url_quote %][% END %]">
+              <img height="50" width="50" src="icons/[%p.id%].png" border="0">
+              </a>&nbsp;
+     </td>
+     </tr>
+     <tr>
+      <td>
         <a href="[% target %]?product=[% p.name FILTER url_quote -%]
               [%- IF cloned_bug_id %]&amp;cloned_bug_id=[% cloned_bug_id FILTER url_quote %][% END -%] 
               [%- IF format %]&amp;format=[% format FILTER url_quote %][% END %]">
         [% p.name FILTER html FILTER no_break %]</a>:&nbsp;
-      </th>
-
-      <td valign="top">[% p.description FILTER html_light %]</td>
+      </td>
+      </tr>
+      <tr>
+      <td>[% p.description FILTER html_light %]</td>
+      </tr>
+      </table>
     </tr>
   [% END %]
 
diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/global/code-error.html.tmpl en/default/global/code-error.html.tmpl
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/global/code-error.html.tmpl	2010-02-06 18:16:43.000000000 +0000
+++ en/default/global/code-error.html.tmpl	2010-08-10 09:16:13.054451052 +0000
@@ -452,6 +452,9 @@
   [% ELSIF error == "unknown_method" %]
     The requested method '[% method FILTER html %]' was not found.
 
+  [% ELSIF error == "unknown_method" %]
+    The requested method '[% method FILTER html %]' was not found.
+
   [% ELSIF error == "usage_mode_invalid" %]
     '[% invalid_usage_mode FILTER html %]' is not a valid usage mode.
 
diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/global/common-links.html.tmpl en/default/global/common-links.html.tmpl
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/global/common-links.html.tmpl	2009-08-13 15:57:22.000000000 +0000
+++ en/default/global/common-links.html.tmpl	2010-08-10 09:16:13.056451103 +0000
@@ -111,7 +111,7 @@
     [% IF doc_section && Param('docs_urlbase') %]
       <li>
         <span class="separator">| </span>
-        <a href="[% docs_urlbase _ doc_section FILTER html %]" target="_blank">Help</a>
+        <a href="http://en.pardus-wiki.org/How:Efficient_Bug_Reporting" target="_blank">Help</a>
       </li>
     [% END %]
 [% END %]
diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/index.html.tmpl en/default/index.html.tmpl
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/index.html.tmpl	2009-10-09 04:39:17.000000000 +0000
+++ en/default/index.html.tmpl	2010-08-10 09:16:13.524326684 +0000
@@ -166,13 +166,9 @@
             </ul>
             <ul class="additional_links">
               <li>
-                <a href="[% docs_urlbase FILTER html %]using.html">
+                <a href="http://en.pardus-wiki.org/How:Efficient_Bug_Reporting">
                   [%- terms.Bugzilla %] User's Guide
                 </a>
-                &nbsp;|
-              </li>
-              <li>
-                <a href="page.cgi?id=release-notes.html" > Release Notes</a>
               </li>
             </ul>
           </div>
diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/list/edit-multiple.html.tmpl en/default/list/edit-multiple.html.tmpl
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/list/edit-multiple.html.tmpl	2009-02-02 18:34:40.000000000 +0000
+++ en/default/list/edit-multiple.html.tmpl	2010-08-10 09:16:13.161324495 +0000
@@ -114,7 +114,7 @@
   </tr>
 
   <tr>
-    <th>
+    <!--th>
       <label for="op_sys">
         <a href="page.cgi?id=fields.html#op_sys">OS</a>:
       </label>
@@ -122,7 +122,7 @@
     <td [% " colspan=\"3\"" IF !Param("usetargetmilestone") %]>
       [% PROCESS selectmenu menuname = "op_sys"
                             menuitems = op_sys %]
-    </td>
+    </td-->
 
     [% IF Param("usetargetmilestone") %]
       <th><label for="target_milestone">Target Milestone:</label></th>
diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/pages/bug-writing.html.tmpl en/default/pages/bug-writing.html.tmpl
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/pages/bug-writing.html.tmpl	2007-08-20 18:25:05.000000000 +0000
+++ en/default/pages/bug-writing.html.tmpl	2010-08-10 09:16:12.883451492 +0000
@@ -74,41 +74,37 @@
     Click the word "Component" to see a description of each 
     component. If none seems appropriate, look for a "General" component.</p>
 
-    <p><b>OS:</b> On which operating system (OS) did you find 
-          it?
-    (e.g. Linux, Windows XP, Mac OS X.)<br>
-    If you know the [% terms.bug %] happens on more than one type of 
-    operating system, choose "All". 
-    If your OS isn't listed, choose Other.</p>
-
     <p><b>Summary:</b> How would you describe the [% terms.bug %], in 
     approximately 60 or fewer characters?<br>
      A good summary should <b>quickly and uniquely identify [% terms.abug %]
     report</b>. It should explain the problem, not your suggested solution.<br>
     <ul>
-     <li>Good: "<tt>Cancelling a File Copy dialog crashes 
+     <li>Good Example: "<tt>Cancelling a File Copy dialog crashes 
      File Manager</tt>"</li>
-     <li>Bad: "<tt>Software crashes</tt>"</li>
-     <li>Bad: "<tt>Browser should work with my web site</tt>"</li>
+     <li>Bad Example: "<tt>Software crashes</tt>"</li>
+     <li>Bad Example: "<tt>Browser should work with my web site</tt>"</li>
     </ul>
     
-    <b>Description:</b>
+    <b>Details:</b>
     The details of your problem report, including:</p>
 
-    <blockquote>
-      <p><b>Overview:</b> More detailed restatement of
-      summary.</p>
+      More detailed restatement of summary.
 
-      <blockquote>
-<pre>
-Drag-selecting any page crashes Mac builds in the NSGetFactory function.
-</pre>
-      </blockquote>
+        <pre>
+        Drag-selecting any page crashes Mac builds in the NSGetFactory function.
+        </pre>
+
+      <p><b>Reporoducibilty</b> You should choose the repeatability frequency of the bug.
+
+      <ul>
+      <li>Happens every time.</li>
+      <li>Happens sometimes, but not always.</li>
+      <li>Haven't tried to reproduce it.</li>
+      <li>Tried, but couldn't reproduce it.</li>
+      </ul>
 
-      <p><b>Steps to Reproduce:</b> Minimized, easy-to-follow steps that
-      will trigger the [% terms.bug %]. Include any special setup steps.</p>
+      <p><b>Steps to Reproduce:</b> Minimized, easy-to-follow steps that  will trigger the [% terms.bug %]. Include any special setup steps.</p>
 
-      <blockquote>
 <pre>
 1) View any web page. (I used the default sample page, 
 resource:/res/samples/test0.html)
@@ -118,59 +114,30 @@
 point in the browser's content region to the bottom of the 
 browser's content region.)
 </pre>
-      </blockquote>
 
-      <p><b>Actual Results:</b> What the application did after performing
-      the above steps.</p>
+      <p><b>Actual Results:</b> What the application did after performing the above steps.</p>
 
-      <blockquote>
 <pre>
 The application crashed.
 </pre>
-      </blockquote>
 
-      <p><b>Expected Results:</b> What the application should have done,
-      were the [% terms.bug %] not present.</p>
+      <p><b>Expected Results:</b> What the application should have done, were the [% terms.bug %] not present.</p>
 
-      <blockquote>
 <pre>
 The window should scroll downwards. Scrolled content should be selected. 
 (Or, at least, the application should not crash.)
 </pre>
-      </blockquote>
-
-      <p><b>Build Date &amp; Platform:</b> Date and platform of the build
-      in which you first encountered the [% terms.bug %].</p>
-
-      <blockquote>
-<pre>
-Build 2006-08-10 on Mac OS 10.4.3
-</pre>
-      </blockquote>
-
-      <p><b>Additional Builds and Platforms:</b> Whether or not 
-      the [% terms.bug %] takes place on other platforms (or browsers, 
-      if applicable).</p>
-
-      <blockquote>
-<pre>
-Doesn't Occur On Build 2006-08-10 on Windows XP Home (Service Pack 2)
-</pre>
-      </blockquote>
 
-      <p><b>Additional Information:</b> Any other useful information.
-      <br><br>For crashing [% terms.bugs %]:</p>
+      <p><b>Additional Information:</b> Any other useful information. 
+      Add any additional information you feel may be relevant to this [% terms.Bug %], such as your system information, stack trace, or a testcase. Any information longer than a few lines, should be      added using the <b>Add an Attachment</b> link on the [% terms.Bug %].
 
-      <ul>
-        <li><b>Windows:</b> Note the type of the crash, and the module that the
-        application crashed in (e.g. access violation in apprunner.exe).</li>
-
-        <li><b>Mac OS X:</b> Attach the "Crash Reporter" log that appears
-        upon crash.
-        Only include the section directly below the crashing thread, usually 
-        titled "Thread 0 Crashed". Please do not paste the entire log!</li>
-      </ul>
-    </blockquote>
+    <p><b>Severity</b> Say how serious the problem is, or if your [% terms.Bug %] is a request for a new feature.
+    <ul>
+        <li>Happens every time.</li>
+        <li>Happens sometimes, but not always.</li>
+        <li>Haven't tried to reproduce it.</li>
+        <li>Tried, but couldn't reproduce it.</li>
+    </ul>
 
   <p>Double-check your report for errors and omissions, then press "Commit".
   Your [% terms.bug %] report will now be in 
diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/reports/duplicates-table.html.tmpl en/default/reports/duplicates-table.html.tmpl
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/reports/duplicates-table.html.tmpl	2007-08-20 18:25:06.000000000 +0000
+++ en/default/reports/duplicates-table.html.tmpl	2010-08-10 09:16:12.773450812 +0000
@@ -56,8 +56,8 @@
                               { name => "component", description => "Component" },
                               { name => "bug_severity", description => "Severity" },
                               { name => "op_sys", description => "Op Sys" },
-                              { name => "target_milestone",
-                                description => "Target<br>Milestone" },
+                              #{ name => "target_milestone",
+                              #  description => "Target<br>Milestone" },
                               { name => "short_desc", description => "Summary" } ]
          %]
 
@@ -133,7 +133,7 @@
 
           <td>[% bug.component FILTER html %]</td>
           <td><center>[% bug.bug_severity FILTER html %]</center></td>
-          <td><center>[% bug.op_sys FILTER html %]</center></td>
+          <!--td><center>[% bug.op_sys FILTER html %]</center></td-->
           <td><center>[% bug.target_milestone FILTER html %]</center></td>
           <td>[% bug.short_desc FILTER html %]</td>
         </tr>
diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/search/boolean-charts.html.tmpl en/default/search/boolean-charts.html.tmpl
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/search/boolean-charts.html.tmpl	2010-03-29 21:32:58.000000000 +0000
+++ en/default/search/boolean-charts.html.tmpl	2010-08-10 09:16:13.104326274 +0000
@@ -44,7 +44,6 @@
   "changedto",
   "changedby",
   "matches",
-  "notmatches",
 ] %]
 
   <p>
diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/search/form.html.tmpl en/default/search/form.html.tmpl
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/search/form.html.tmpl	2010-02-08 00:19:54.000000000 +0000
+++ en/default/search/form.html.tmpl	2010-08-10 09:16:13.104326274 +0000
@@ -153,7 +153,6 @@
     <td colspan="4">
       <table>
         <tr>
-        [% Hook.process('before_selects_top') %]
         [% IF Param('useclassification') %]
           <td valign="top">
             <table>
@@ -253,7 +252,6 @@
             </table>
           </td>
         [% END %]
-        [% Hook.process('after_selects_top') %]
         </tr>
       </table>
     </td>
@@ -261,24 +259,18 @@
 
 [%# *** Comment URL Whiteboard Keywords *** %]
 
-  [% SET freetext_fields = [
+  [% FOREACH field = [
     { name => "longdesc", description => "A&nbsp;<u>C</u>omment",
       accesskey => 'c' },
     { name => "bug_file_loc", description => "The&nbsp;<u>U</u>RL",
       accesskey => 'u' },
     { name => "status_whiteboard", description => "<u>W</u>hiteboard",
-      accesskey => 'w' } 
-  ] %]
+      accesskey => 'w' } ] %]
 
-  [% Hook.process('before_freetext_fields') %]
-  [% FOREACH field = freetext_fields %]
-    [% NEXT IF field.name == 'status_whiteboard' 
-               AND NOT Param('usestatuswhiteboard') 
-    %]
+    [% UNLESS field.name == 'status_whiteboard' AND NOT Param('usestatuswhiteboard') %]
     <tr>
       <th align="right">
-        <label for="[% field.name %]" 
-               accesskey="[% field.accesskey %]">[% field.description %]</label>:
+        <label for="[% field.name %]" accesskey="[% field.accesskey %]">[% field.description %]</label>:
       </th>
       <td>
         [% type = field.name _ "_type" %]
@@ -291,9 +283,10 @@
       </td>
       <td></td>
     </tr>
+    [% END %]
   [% END %]
 
-  [% IF use_keywords %]
+  [% IF have_keywords %]
     <tr>
       <th align="right">
         <label for="keywords" accesskey="k"><a href="describekeywords.cgi"><u>K</u>eywords</a></label>:
@@ -301,7 +294,7 @@
       <td>
         [% INCLUDE "search/type-select.html.tmpl"
            name = "keywords_type",
-           types = ['allwords', 'anywords', 'nowords', 'regexp', 'notregexp'],
+           types = ['allwords', 'anywords', 'nowords'], 
            selected = default.keywords_type.0 %]
       </td>
       <td>
@@ -312,7 +305,7 @@
   [% END %]
 
   [%# Deadline %]
-  [% IF user.is_timetracker %]
+  [% IF user.in_group(Param("timetrackinggroup")) %]
     <tr>
       <th align="right">
         <label for="deadlinefrom" accesskey="l">Dead<u>l</u>ine</label>:
@@ -324,12 +317,10 @@
                   value="[% default.deadlineto.0 FILTER html %]">
       </td>
       <td>
-        <small>(YYYY-MM-DD or relative dates)</small>
+        <small>(YYYY-MM-DD)</small>
       </td>
     </tr>
   [% END %]
-
-  [% Hook.process('after_freetext_fields') %]
   
 </table>
 
@@ -339,7 +330,6 @@
 
 <table>
   <tr>
-    [% Hook.process('before_selects_bottom') %]
     <td>
       <table>
         <tr>
@@ -405,7 +395,7 @@
         </tr>
       </table>
     </td>
-    <td>
+    <!--td>
       <table>
         <tr>
           <th align="left">
@@ -417,8 +407,7 @@
                                     size => 7 } %]
         </tr>
       </table>
-    </td>
-    [% Hook.process('after_selects_bottom') %]
+    </td-->
   </tr>
 </table>
 
@@ -535,9 +524,9 @@
 <table>
   <tr>
     <td>
-      <select name="bug_id_type">
-        <option value="anyexact"[% " selected" IF default.bug_id_type.0 == "anyexact" %]>Only include</option>
-        <option value="nowords"[% " selected" IF default.bug_id_type.0 == "nowords" %]>Exclude</option>
+      <select name="bugidtype">
+        <option value="include"[% " selected" IF default.bugidtype.0 == "include" %]>Only include</option>
+        <option value="exclude"[% " selected" IF default.bugidtype.0 == "exclude" %]>Exclude</option>
       </select>
       <label for="bug_id">[% terms.bugs %] numbered</label>:
     </td>
@@ -623,20 +612,17 @@
   <td align="left">
     <select name="[% sel.name %]" id="[% sel.name %]"
             multiple="multiple" size="[% sel.size %]">
-      [% FOREACH value = ${sel.name} %]
-        [% IF value.id %]
-          [%# This only applies for Resolution really %]
-          [% v = value.name OR '---' -%]
-          <option value="[% v FILTER html %]"
-            [% " selected" IF lsearch(default.${sel.name}, v) != -1 %]>
-            [% display_value(sel.name, value.name) FILTER html %]
-          </option>
-        [% ELSE %]
-          <option value="[% value OR '---' FILTER html %]"
-            [% " selected" IF lsearch(default.${sel.name}, value) != -1 %]>
-            [% display_value(sel.name, value) FILTER html %]
-          </option>
-        [% END %]
+      [% FOREACH name = ${sel.name} %]
+        <option value="[% name FILTER html %]"
+          [% " selected" IF lsearch(default.${sel.name}, name) != -1 %]>
+          [% IF sel.name == "bug_status" %]
+            [% get_status(name) FILTER html %]
+          [% ELSIF sel.name == "resolution" %]
+            [% get_resolution(name) FILTER html %]
+          [% ELSE %]
+            [% name FILTER html %]
+          [% END %]
+        </option>
       [% END %]
     </select>
   </td>
diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/search/search-specific.html.tmpl en/default/search/search-specific.html.tmpl
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/search/search-specific.html.tmpl	2009-07-20 04:17:44.000000000 +0000
+++ en/default/search/search-specific.html.tmpl	2010-08-10 09:16:13.105324785 +0000
@@ -21,7 +21,7 @@
 [% PROCESS global/variables.none.tmpl %]
 
 [% PROCESS global/header.html.tmpl 
-  title = "Simple Search"
+  title = "Find a Specific " _ terms.Bug
 %]
 
 [% WRAPPER search/tabs.html.tmpl %]
diff -Nuar --exclude='\.svn' /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/search/tabs.html.tmpl en/default/search/tabs.html.tmpl
--- /home/pars/İndirilenler/bugzilla-3.4.6/template/en/default/search/tabs.html.tmpl	2009-07-20 04:17:44.000000000 +0000
+++ en/default/search/tabs.html.tmpl	2010-08-10 09:16:13.106326300 +0000
@@ -24,7 +24,7 @@
   #%]
 
 [% WRAPPER global/tabs.html.tmpl
-     tabs = [ { name => 'specific', label => "Simple Search",
+     tabs = [ { name => 'specific', label => "Find a Specific $terms.Bug",
                 link => "query.cgi?format=specific" },
               { name => 'advanced', label => "Advanced Search",
                 link => "query.cgi?format=advanced" } ]
