Index: src/maker.py
===================================================================
--- src/maker.py	(revision 17718)
+++ src/maker.py	(working copy)
@@ -110,6 +110,40 @@
     # Generate the config file
     generate_grub_conf(project, kernel, initramfs)
 
+def setup_isolinux(project):
+    print "Generating isolinux files..."
+    xterm_title("Generating isolinux files")
+
+    image_dir = project.image_dir()
+    iso_dir = project.iso_dir()
+    kernel = ""
+    initramfs = ""
+
+    # Setup dir
+    path = os.path.join(iso_dir, "boot/isolinux")
+    if not os.path.exists(path):
+        os.makedirs(path)
+
+    def copy(src, dest):
+        run('cp -P "%s" "%s"' % (src, os.path.join(iso_dir, dest)))
+
+    # Copy the kernel and initramfs
+    path = os.path.join(image_dir, "boot")
+    for name in os.listdir(path):
+        if name.startswith("kernel") or name.startswith("initramfs") or name.endswith(".bin"):
+            if name.startswith("kernel"):
+                copy(os.path.join(path, name), "boot/kernel")
+            elif name.startswith("initramfs"):
+                copy(os.path.join(path, name), "boot/initrd")
+
+    # copy config and gfxboot stuff
+    tmplpath = os.path.join(image_dir, "usr/lib/syslinux/pardus")
+    dest = os.path.join(iso_dir, "boot/isolinux")
+    for name in os.listdir(tmplpath):
+        copy(os.path.join(tmplpath, name), dest)
+
+    copy(os.path.join("usr/lib/syslinux/isolinux-debug.bin", dest))
+
 #
 # Image related stuff
 #
@@ -306,12 +340,14 @@
                 if name != ".svn":
                     copy(os.path.join(path, name), name)
         
-        setup_grub(project)
+        # setup_grub(project)
+        setup_isolinux(project)
         
         if project.type == "install":
             run('ln -s "%s" "%s"' % (project.install_repo_dir(), os.path.join(iso_dir, "repo")))
         
-        run('mkisofs -f -J -joliet-long -R -l -V "Pardus" -o "%s" -b boot/grub/stage2_eltorito -no-emul-boot -boot-load-size 4 -boot-info-table "%s"' % (
+        # run('mkisofs -f -J -joliet-long -R -l -V "Pardus" -o "%s" -b boot/grub/stage2_eltorito -no-emul-boot -boot-load-size 4 -boot-info-table "%s"' % (
+        run('mkisofs -f -J -joliet-long -R -l -V "Pardus" -o "%s" -b boot/isolinux/isolinux.bin -c boot/isolinux/boot.cat -no-emul-boot -boot-load-size 4 -boot-info-table "%s"' % (
             iso_file,
             iso_dir,
         ))
