Index: src/maker.py
===================================================================
--- src/maker.py	(revision 18529)
+++ src/maker.py	(working copy)
@@ -113,6 +113,42 @@
     # Generate the config file
     generate_grub_conf(project, kernel, initramfs)
 
+def generate_isolinux_conf(project):
+    print "Generating isolinux config files..."
+    xterm_title("Generating isolinux config files")
+
+    dict = {}
+    dict["title"] = project.title
+    dict["exparams"] = project.exparams or ''
+
+    image_dir = project.image_dir()
+    iso_dir = project.iso_dir()
+
+    isolinux_tmpl = """
+prompt 1
+timeout 200
+
+gfxboot /boot/isolinux/init
+
+label %(title)s
+    kernel /boot/kernel
+    append initrd=/boot/initrd root=/dev/ram0 vga=791 splash=silent quiet %(exparams)s
+
+label harddisk
+    localboot 0x80
+
+label memtest
+    kernel /boot/memtest
+"""
+
+    dest = os.path.join(iso_dir, "boot/isolinux/isolinux.cfg")
+    data = isolinux_tmpl % dict
+
+    f = file(dest, "w")
+    f.write(data % dict)
+    f.close()
+
+
 def setup_isolinux(project):
     print "Generating isolinux files..."
     xterm_title("Generating isolinux files")
@@ -140,13 +176,16 @@
                 copy(os.path.join(path, name), "boot/initrd")
 
     # copy config and gfxboot stuff
-    tmplpath = os.path.join(image_dir, "usr/lib/syslinux/pardus")
+    generate_isolinux_conf(project)
+
+    tmplpath = os.path.join(image_dir, "usr/share/gfxtheme/pardus/install")
     dest = os.path.join(iso_dir, "boot/isolinux")
     for name in os.listdir(tmplpath):
         copy(os.path.join(tmplpath, name), dest)
 
     copy(os.path.join(image_dir, "usr/lib/syslinux/isolinux-debug.bin"), "%s/isolinux.bin" % dest)
 
+
 #
 # Image related stuff
 #
